### PR TITLE
Svg titles and fillColor prop

### PR DESCRIPTION
**Pull request by kristastarr**
Updates to the `Idea` icon which were prioritized because they are being used in production right now and the removal of the svg `<title>` is necessary... these changes will be applied to additional icons in the future 

- Refactored `IdeaMotion` component to reduce amount of code 
- Added optional prop `needsTitle`
  - If `needsTitle` is `false`, or if a prop is not provided (because the developer plans to provide naming for icon via other means) then there will be no `<title>` element in the svg
  - If `needsTitle` is `true`, a custom title can be specified using the optional `customTitle` prop.  If `needsTitle` is set to `true` and no `customTitle` prop is provided, the default title will be the icon's name as indicated in Carbon 
  - Added helper function to return the correct svg title based on the props
 - Added optional prop `fillColor` 
   - If a 6-digit hex code is provided via the `fillColor` prop, that will be the color of the icon.  If none is provided, it will be G100 (`#161616`)
   - In future, I will try to make it work with Carbon tokens, but for now only can get it to work without the `#` 
 - Added PropTypes for all props including new ones
 - Added default values for all props

### CODE REVIEW

**Comment by kristastarr** [comment]
@elycheea @glapadre Could one of y'all review this when you have time? 🙏

**Comment by elycheea** [comment]
@kristastarr Don’t forget to look into the conflict!

**Review by tay1orjones — APPROVED** [review]
+1 what Elysia said on removing `needsTitle` to instead infer if a default title should be given based on if `customTitle` is provided. Also you might consider renaming `customTitle` to just be `title`.

One other minor comment below, otherwise looks good to me.

File: src/components/Operations/IdeaMotion/IdeaMotion.js [hunk 1]
```diff
@@ -1,48 +1,69 @@
...
-  
 
-export default IdeaMotion
+ IdeaMotion.defaultProps = {
+    isAnimating: false,
+    size: 32,
+    needsTitle: false,
+    customTitle: null,
+    fillColor: "161616",
+  }
```
[the remark this answers is not in the document]
Comment (kristastarr) [reply src/components/Operations/IdeaMotion/IdeaMotion.js]: Thank you! I will take a look at it

**Comment by kristastarr** [comment]
@elycheea @tay1orjones I made the update to require one `title` prop instead of two separate ones for `needsTitle` and `customTitle`.  Updated propTypes as well.  Now there is a default title (the icon's name) if not otherwise specified.  
I don't think this is the most eloquent solution but it's working. Is it okay with y'all if I go ahead and merge it, and can improve on it later?

**Comment by kristastarr** [comment]
@elycheea @tay1orjones I made the update to require one `title` prop instead of two separate ones for `needsTitle` and `customTitle`.  Updated propTypes as well.  Now there is a default title (the icon's name) if not otherwise specified.  
I don't think this is the most eloquent solution but it's working. Is it okay with y'all if I go ahead and merge it, and can improve on it later?

**Pull request merged**